kad/src/protocol.rs

```diff
@@ -262,13 +272,22 @@ fn proto_to_msg(mut message: protobuf_structs::dht::Message) -> Result<KadMsg, I
                 Ok(KadMsg::FindNodeReq {
                     key: message.take_key(),
                 })
+
             } else {
+                let mut closer_peers = Vec::with_capacity(message.get_closerPeers().len());
```

Comment:
Why not:

```rust
message.mut_closerPeers()
    .iter_mut()
    .filter_map(|peer| Peer::from_peer(peer).ok())
    .collect::<Vec<_>>()
```

Is the possible increased number of allocations really worth it?